GitCommands/PathUtil.cs

```diff
@@ -124,14 +124,42 @@ public static string NormalizePath([NotNull] this string path)
 
             try
             {
-                return Path.GetFullPath(new Uri(path).LocalPath);
+                return Path.GetFullPath(Resolve(path));
             }
             catch (UriFormatException)
             {
                 return string.Empty;
             }
         }
 
+        [NotNull]
+        public static string Resolve([NotNull] string path, string relativePath = "")
+        {
+            if (string.IsNullOrWhiteSpace(path))
+            {
+                throw new ArgumentException(nameof(path));
+            }
+
+            // Fix for on purpose invalid machine name of WSL in Windows 10
+            bool replaceWslPath = path.StartsWith(@"\\wsl$\");
```

Comment:
This literal constant is used in several places. If it is not case-sensitive, create a proper const and reuse it